src/js/core/index.js

```diff
@@ -33,26 +33,31 @@ const noExport = [
     'evalFile',
     'evalScript',
     'ffi_load_native',
+    'getPid',
+    'getPpid',
     'guessHandle',
     'isStdinTty',
+    'isWorker',
```

Comment:
Making this not visible, is there an alternative web compliant mechanism working to detect if it is a worker?

Reply:
Yes, self.postMessage for example. But also, I should stop defining self on the main, self only has meaning in a worker scope.